### ISSUE TITLE
Theme: Tree hierarchy in Virtualmin site dropdown

### ISSUE THREAD

**Issue by bilekt3** [qooob/authentic-theme#466]
It will be nice and helpful to see tree hierarchy in virtualmin server list dropdown. Flat hierarchy is little confusing.

**Comment by qooob** [comment]
You mean on the navigation menu? Top right? Why select is not good?

**Comment by bilekt3** [comment]
I mean this dropdown. 
<img width="263" alt="screen shot 2016-05-25 at 15 11 14" src="https://cloud.githubusercontent.com/assets/381740/15541219/bb5dde54-228b-11e6-9b76-6c01f119042b.png">

In old theme this dropdown respect hierarchy of servers, subserves and aliases. Now in flat structure is very confusing what belongs where. It was achieved with spaces before server name if I remember correctly.

Quick search is excellent but some sort of visualization of hierarchy might speedup my workflow. In example if I am going to find some sort of sub-server I would like to be informed about his master server somehow.

It is more clear now?

**Comment by qooob** [comment]
A bit more, clear. Thanks. But as far as I remember, there is padding on the left for subservers already? Could you please post a screenshot of how it looks in the old theme and how you would like it to be?

**Comment by bilekt3** [comment]
Yes you remember correctly. But it is weird drop down is not on my system anymore on old theme. Reason why I am interested in this is that I have 160+ domains in my virtualmin stack.

If I am thinking about improvement that indication is very tricky because filtering and spaces are not good  together.

One (not only) solution might be to indicate position with spaces (icon?) and when filter is applied maintain server hierarchy i.e.: Given server list

- - Server A
- - - Subserver AA
- - - Subserver AB
- - Server B

When I start filter for example **Subserver AB** whole structure remains in filter:

- - Server A
- - - Subserver AA
- - - Subserver AB

Any other ideas?

**Comment by qooob** [comment]
I'll see what I can do in few hours. It might be difficult to implement.. But will see.

**Comment by bilekt3** [comment]
👍 Thanks for considering this. As I said it is tricky but it might be worth it. I am surprised that nobody have similar problem.

**Comment by qooob** [comment]
Take a look at the screenshot. It has padding for sub domains, like I mentioned.

![screenshot from 2016-05-25 17-41-17](https://cloud.githubusercontent.com/assets/4426533/15544057/65f409b2-22a8-11e6-8eaa-99a5c277392f.png)

At least it's there at upcoming 18.00. Is this good enough?

**Comment by bilekt3** [comment]
It is super, please note, that there is also alias (for server also sub-server). Main server view is excellent (red one).

**Comment by qooob** [comment]
Ok, what exactly is missing?

**Comment by qooob** [comment]
I mean there is search? What else would you want?

A
  abc.com
  abc1.com
  abc2.com
B
   bcd.com
   bcd1.com
   bcd2.com

This kind of stuff??

**Comment by bilekt3** [comment]
Sorry for late response.

My note was about indentation based on real position in tree structure.

As far I know you can do following server types in Virtualmin:

1) Server (root, in virtualmin referenced as Top server)
2) Alias to Server (two indents for root)
3) Sub-Server (one indent for root, in virtualmin referenced as sub-server)
4) Alias of sub-server (one indent for sub-server, two indents for Server)
5) Alias of sub-server with own mail (for me is same as 4)

```
A.com (Top server)
-- Alias AA.com (for top-server A.com)
- Sub-server AB.com
-- Alias ABB.com (for subserver AB.com)
B.com (Top server)
-- Alias BA.com (for  top-server B.com)
```

**Comment by qooob** [comment]
It will be implemented on 18.20

**Comment by qooob** [comment]
Version 18.20 is out, please upgrade.
